e2e_tests/tests/test_password_reset.py

```diff
@@ -0,0 +1,377 @@
+#
+"""
+Test password reset flow for Ocim console.
+"""
+from collections import namedtuple
+
+from django.conf import settings
+from django.core import mail as django_mail
+from django_rest_passwordreset.models import ResetPasswordToken
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.common.exceptions import NoSuchElementException
+from simple_email_confirmation.models import EmailAddress
+
+from e2e_tests.tests.utils import BrowserLiveServerTestCase
+from instance.tests.base import create_user_and_profile
+from registration.models import BetaTestApplication
+
+
+class ForgotPasswordTestCase(BrowserLiveServerTestCase):
+    """
+    Tests the forgot passwor page.
+    """
+
+    def setUp(self):
+        """
+        Setup the user with a BetaTestApplication
+        """
+        super().setUp()
+        self.user_with_app = create_user_and_profile('jerry.mouse', 'jerry.mouse@example.com')
+        self.application = BetaTestApplication.objects.create(
+            user=self.user_with_app,
+            subdomain="cheese-school",
+            instance_name="Jerry's School of Cheese",
+            public_contact_email="jerry.mouse@example.com",
+            privacy_policy_url="http://www.some/url"
+        )
+        EmailAddress.objects.create_confirmed(
+            email=self.application.public_contact_email,
+            user=self.user_with_app,
+        )
+        self.forgot_password_url = f"{settings.USER_CONSOLE_FRONTEND_URL}/password-forgotten"
+
+    def _open_forgot_password_page(self):
+        """
+        Opens the forgot password page and returns form
+        WebElements
+        """
+        self.browser.get(self.forgot_password_url)
+        self.wait.until(
+            EC.presence_of_element_located((By.TAG_NAME, 'form'))
+        )
+
+        email_field = self.form.find_element_by_name("email")
+        reset_password_btn = self.form.find_element_by_tag_name("button")
+        return email_field, reset_password_btn
+
+    @property
+    def form(self):
+        """
+        Get the form element on the page
+        """
+        return self.browser.find_element_by_tag_name("form")
+
+    def test_forgot_password_page_loads_successfully(self):
+        """
+        Tests that the reset password page loads correctly
+        """
+        # Jerry visits the forgot password page
+        email_field, reset_password_btn = self._open_forgot_password_page()
+
+        # Jerry can see a email input and a reset password button
+        self.assertTrue(email_field.is_displayed())
+        self.assertTrue(reset_password_btn.is_displayed())
+        # Text of the button is "reset password"
+        self.assertEqual(reset_password_btn.text.lower(), "reset password")
+        # Text of the button is in uppercase
+        self.assertEqual(reset_password_btn.value_of_css_property('text-transform'), 'uppercase')
```

Comment:
I think this is testing too much. I don't see how the case of this text is important to its functioning. Please remove this line.